x-pack/plugins/upgrade_assistant/public/application/components/kibana_deprecations/kibana_deprecations_table.tsx

```diff
@@ -0,0 +1,246 @@
+    {
+      defaultMessage: 'Type',
+    }
+  ),
+  resolutionColumnTitle: i18n.translate(
+    'xpack.upgradeAssistant.kibanaDeprecations.table.resolutionColumnTitle',
+    {
+      defaultMessage: 'Resolution',
+    }
+  ),
+  configDeprecationTypeCellLabel: i18n.translate(
+    'xpack.upgradeAssistant.kibanaDeprecations.table.configDeprecationTypeCellLabel',
+    {
+      defaultMessage: 'Config',
+    }
+  ),
+  featureDeprecationTypeCellLabel: i18n.translate(
+    'xpack.upgradeAssistant.kibanaDeprecations.table.featureDeprecationTypeCellLabel',
+    {
+      defaultMessage: 'Feature',
+    }
+  ),
+  unknownDeprecationTypeCellLabel: i18n.translate(
+    'xpack.upgradeAssistant.kibanaDeprecations.table.unknownDeprecationTypeCellLabel',
+    {
+      defaultMessage: 'Uncategorized',
+    }
+  ),
+  typeFilterLabel: i18n.translate(
+    'xpack.upgradeAssistant.kibanaDeprecations.table.typeFilterLabel',
+    {
+      defaultMessage: 'Type',
+    }
+  ),
+  criticalFilterLabel: i18n.translate(
+    'xpack.upgradeAssistant.kibanaDeprecations.table.criticalFilterLabel',
+    {
+      defaultMessage: 'Critical',
+    }
+  ),
+  criticalBadgeLabel: i18n.translate(
+    'xpack.upgradeAssistant.kibanaDeprecations.table.criticalBadgeLabel',
+    {
+      defaultMessage: 'critical',
+    }
+  ),
+  searchPlaceholderLabel: i18n.translate(
+    'xpack.upgradeAssistant.kibanaDeprecations.table.searchPlaceholderLabel',
+    {
+      defaultMessage: 'Filter',
+    }
+  ),
+};
+
+interface Props {
+  deprecations?: KibanaDeprecationDetails[];
+  reload: () => void;
+  toggleFlyout: (newFlyoutContent?: KibanaDeprecationDetails) => void;
+  deprecationResolutionState?: DeprecationResolutionState;
+}
+
+export const KibanaDeprecationsTable: React.FunctionComponent<Props> = ({
+  deprecations,
+  reload,
+  toggleFlyout,
+  deprecationResolutionState,
+}) => {
+  const columns: Array<EuiBasicTableColumn<KibanaDeprecationDetails>> = [
+    {
+      field: 'level',
+      name: i18nTexts.statusColumnTitle,
+      width: '5%',
+      truncateText: true,
+      sortable: true,
+      render: (level: KibanaDeprecationDetails['level']) => {
+        if (level === 'critical') {
+          return <EuiBadge color="danger">{i18nTexts.criticalBadgeLabel}</EuiBadge>;
+        }
+
+        return <>{''}</>;
+      },
+    },
+    {
+      field: 'title',
+      width: '40%',
+      name: i18nTexts.issueColumnTitle,
+      truncateText: true,
+      sortable: true,
+      render: (title: KibanaDeprecationDetails['title'], deprecation: KibanaDeprecationDetails) => {
+        return (
+          <EuiLink
+            onClick={() => toggleFlyout(deprecation)}
+            data-test-subj="deprecationDetailsLink"
+          >
+            {title}
+          </EuiLink>
+        );
+      },
+    },
+    {
+      field: 'deprecationType',
+      name: i18nTexts.typeColumnTitle,
+      width: '20%',
+      truncateText: true,
+      sortable: true,
```

Comment:
Not sure why but when I have just "Config" and "Uncategorized" rows, the order remains the same regardless of which direction I sort on this column.

![image](https://user-images.githubusercontent.com/1238659/131757894-440955b5-1d1e-4fb5-877d-1a6953caaf63.png)

Reply:
Good catch! Fixed.

Reply:
The implementation will also be improved once https://github.com/elastic/kibana/issues/111070 is addressed.